planner/core/task.go

```diff
@@ -1868,84 +2305,35 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
 	case MppScalar:
 		prop := &property.PhysicalProperty{TaskTp: property.MppTaskType, ExpectedCnt: math.MaxFloat64, MPPPartitionTp: property.SinglePartitionType}
 		if !mpp.needEnforceExchanger(prop) {
+			// one hand: when the low layer already satisfied the single partition layout, just do the all agg computation in the single node.
 			return p.attach2TaskForMpp1Phase(mpp)
 		}
+		// other hand: try to split the mppScalar agg into multi phases agg **down** to multi nodes since data already distributed across nodes.
```

Comment:
```suggestion
		// On the other hand: try to split the mppScalar agg into multi phases agg **down** to multi nodes since data are already distributed across nodes.
```